js/api/Geofencing.js

```diff
@@ -0,0 +1,200 @@
+import { InteractionManager, Platform } from 'react-native'
+import _ from 'lodash'
+import Parse from 'parse/react-native'
+import Store from '../data/Store'
+import realm from '../data/Realm'
+import PushNotification from 'react-native-push-notification';
+
+import { BackgroundGeolocation } from './BackgroundProcess'
+import { loadAllCachedGeofenceTriggers } from './Triggers'
+import { showToast } from './Notifications'
+
+
+let activeMap; // Cache a MapPage component for to update when geofencing triggers are crossed.
+
+/**
+ * Caches a MapPage component inside a variable for easy access
+ * @param {element} component     React class element to be cached
+ */
+export function setActiveMap(component) {
+  activeMap = component
+  connectMapToGeofence()
+}
+
+/**
+ * Clears and deactivates the currently cached map element
+ * @param  {element} component    React class element to be cleared
+ */
+export function clearActiveMap(component) {
+  if (activeMap == component) {
+    component.active = false
+    activeMap = null
+  }
+}
+
+/**
+ * Creates native geofence objects using the 3rd-party geolocation library.
+ * Erases previous set of native geofences and adds new ones based on geofence triggers.
+ */
+export function setupGeofences() {
+  // BackgroundGeolocation.stop()
+
+  getUserLocationData((response) => {console.log(response)})
+
+  loadAllCachedGeofenceTriggers({excludeCompleted: true}, (err, response) => {
+    resetGeofences((err) => {
+      if (err) {
+        console.warn('Unable to load geofence triggers');
+        console.warn(err);
+        return;
+      }
+
+      const triggerGeofences = response.map((trigger) => {
+        return {
+          identifier: trigger.id,
+          radius: trigger.radius,
+          latitude: trigger.latitude,
+          longitude: trigger.longitude,
+          notifyOnEntry: true,
+          notifyOnExit: true,
+          notifyOnDwell: true,
+          loiteringDelay: 30000
+        }
+      })
+      
+      console.log('Adding ' + triggerGeofences.length + ' geofences...')
+      BackgroundGeolocation.addGeofences(triggerGeofences, function() {
+        try {
```

Comment:
Another left-over from debugging, thank you for catching that.
This was before I upgraded to new versions of react-native to get the native geofencing modules behaving properly.

Fixed.